cli/ioctl/cmd/action/actiontransfer.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/iotexproject/iotex-core/pkg/keypair"
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/protogen/iotexapi"
+	"github.com/iotexproject/iotex-core/protogen/iotextypes"
+)
+
+// actionTransferCmd transfers tokens on IoTex blockchain
+var actionTransferCmd = &cobra.Command{
+	Use:   "transfer recipient amount data",
+	Short: "Transfer tokens on IoTex blokchain",
+	Args:  cobra.ExactArgs(3),
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println(transfer(args))
+	},
+}
+
+func init() {
+	actionTransferCmd.Flags().Uint64VarP(&gasLimit, "gas-limit", "l", 0, "set gas limit")
+	actionTransferCmd.Flags().Int64VarP(&gasPrice, "gas-price", "p", 0, "set gas prize")
+	actionTransferCmd.Flags().StringVarP(&alias, "alias", "a", "", "choose signing key")
+	actionTransferCmd.MarkFlagRequired("gas-limit")
+	actionTransferCmd.MarkFlagRequired("gas-price")
+	actionTransferCmd.MarkFlagRequired("alias")
+}
+
+// transfer transfers tokens on IoTex blockchain
+func transfer(args []string) string {
+	recipient := args[0]
+	amount, err := strconv.ParseInt(args[1], 10, 64)
+	if err != nil {
+		log.L().Error("cannot transfer "+args[1]+" into int64", zap.Error(err))
+		return err.Error()
+	}
+	payload := args[2]
+	// TODO: Check the validity of args
+
+	sender, err := account.AliasToAddress(alias)
+	if err != nil {
+		return err.Error()
+	}
+	accountMeta, err := account.GetAccountMeta(sender)
+	if err != nil {
+		log.L().Error("cannot get account from "+recipient, zap.Error(err))
+		return err.Error()
+	}
+	tx, err := action.NewTransfer(accountMeta.PendingNonce, big.NewInt(amount),
+		recipient, []byte(payload), gasLimit, big.NewInt(gasPrice))
+	if err != nil {
+		log.L().Error("cannot make a Transfer instance", zap.Error(err))
+	}
+	fmt.Printf("Enter password #%s:", alias)
+	bytePassword, err := terminal.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		log.L().Error("fail to get password", zap.Error(err))
+		return err.Error()
+	}
+	password := strings.TrimSpace(string(bytePassword))
+	bd := &action.EnvelopeBuilder{}
+	elp := bd.SetNonce(accountMeta.PendingNonce).
+		SetGasPrice(big.NewInt(gasPrice)).
+		SetGasLimit(gasLimit).
+		SetAction(tx).Build()
+	hash := elp.Hash()
+	sig, err := account.Sign(alias, string(password), hash[:])
+	if err != nil {
+		log.L().Error("fail to sign", zap.Error(err))
+		return err.Error()
+	}
+	pubKey, err := crypto.SigToPub(hash[:], sig)
+	selp := &iotextypes.Action{
+		Core:         elp.Proto(),
+		SenderPubKey: keypair.PublicKeyToBytes(pubKey),
+		Signature:    sig,
+	}
+	if err != nil {
+		log.L().Error("fail to sign envelope", zap.Error(err))
+		return err.Error()
+	}
+	request := iotexapi.SendActionRequest{Action: selp}
+
+	endpoint := config.GetEndpoint()
+	if endpoint == config.ErrEmptyEndpoint {
+		log.L().Error(config.ErrEmptyEndpoint)
+		return "use \"ioctl config set endpoint\" to config endpoint first."
+	}
+	conn, err := grpc.Dial(endpoint, grpc.WithInsecure())
```

Comment:
dial in and send an action can be extracted at to a util function, which will be used by many cmds.